arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/WcpOperation.java

```diff
@@ -201,7 +212,42 @@ void trace(Trace trace, int stamp) {
     }
   }
 
-  int maxCt() {
-    return this.isOneLineInstruction ? 1 : LIMB_SIZE;
+  private boolean isOli() {
+    return switch (this.opCode) {
+      case ISZERObv, EQbv -> true;
+      case SLTbv, SGTbv, LTbv, GTbv, LEQbv, GEQbv -> false;
+      default -> throw new IllegalStateException("Unexpected value: " + this.opCode);
+    };
+  }
+
+  private boolean isVli() {
+    return switch (this.opCode) {
+      case LTbv, GTbv, LEQbv, GEQbv, SLTbv, SGTbv -> true;
+      case ISZERObv, EQbv -> false;
+      default -> throw new IllegalStateException("Unexpected value: " + this.opCode);
+    };
+  }
+
+  private int maxCt() {
+    switch (this.opCode) {
+      case ISZERObv, EQbv -> {
+        return 0;
+      }
+      case LTbv, GTbv, LEQbv, GEQbv, SLTbv, SGTbv -> {
+        if (this.arg1.isZero() && this.arg2.isZero()) {
+          return 0;
+        } else {
+          final ArrayList<Integer> sizes = new ArrayList<>(6);
+          sizes.add(this.arg1.slice(0, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.arg2.slice(0, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.arg1.slice(LLARGE, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.arg2.slice(LLARGE, LLARGE).trimLeadingZeros().size());
+          sizes.add(this.adjHi.trimLeadingZeros().size());
+          sizes.add(this.adjLo.trimLeadingZeros().size());
+          return Collections.max(sizes) - 1;
```

Comment:
Is it ever possible that max(sizes) returns 0 E.g. what happens if all arguments to the instruction are 0 ?

Reply:
Oh my bad, you consider that case above :slightly_smiling_face: